swan-lake/development-tutorials/configurability/provide-values-to-configurable-variables.md

```diff
@@ -1,67 +1,137 @@
+| string         | <code>configurable string name = ?; </code>                                                                                                                                           | `$ bal run -- -Cname=John`                                          |
+| xml            | <code>configurable xml book = ?; </code>                                                                                                                                              | `$ bal run -- -CxmlVar="<book>The Lost World</book>"`               |
+| enum           | <code>enum Country { </code><br>    <code>LK = "Sri Lanka", </code><br>    <code>US = "United States" </code><br> <code>} </code><br> <code>configurable Country country = ?; </code> | `$ bal run -- -Ccountry="Sri Lanka"`                                |
+| union          | <code>configurable float&#124;int&#124;string measurement = ?; </code>                                                                                                                | `$ bal run -- -Cmeasurement=5.0`                                    |
+
+## Provide via configuration files
+
+You can provide configuration values via one or more configuration files as follows.
+
+1. Create a `Config.toml` file within your Ballerina project to specify the configurable values complying with the [TOML syntax](#toml-syntax). By default, this will be used when a Ballerina program is executed using the `bal run` command.
+
+2. If you need to use another configuration file, specify its path via the `BAL_CONFIG_FILES` environment variable. 
+
+3. Ballerina also supports specifying multiple configuration files using the `BAL_CONFIG_FILES` environment variable with the OS-specific separator. The file precedence order will be as specified in the environment variable.
+
+    >**Note:** If an environment variable is not specified, a file named `Config.toml` will be sought in the current working directory by default. Once the environment variable is specified, the `Config.toml` will not be considered for the configuration values by default. Therefore, if you are required to use the `Config.toml` file along with others, you need to specify all of them via `BAL_CONFIG_FILES` in the order in which they should be executed. 
+
+4. Configuration values for testing can be provided in a file named `Config.toml` located in the `tests` directory. For more details, see [Define test-specific configurations](/learn/test-ballerina-code/configure-tests/#define-test-specific-configurations).
+
+For example, if the configurable variables are defined in the following way,
+
+```ballerina
+configurable int port = 9090;
+configurable float maxPayload = ?;
+configurable string username = ?;
+configurable boolean verbose = true
+```
+
+and the values are provided in two `Config.toml` and `info.toml` files as follows,
```

Comment:
Isn't it better to end with full stop here?

Reply:
Updated.